### PR TITLE
feat(avatar): Use the first emoji of room names as avatar emoji

### LINKED ISSUE #8444: Ability to set an emoji as a conversation avatar
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**
Whatsapp allows to set an emoji as a group chat icon. I think this could be a very nice UX addition for 2 reasons:
1. facilitate quick group chat creation (without having to choose a picture which takes longer)
2. make easily recognisable group chats (without having to sacrifice the precious characters in the title to make it start with an emoji)

**Describe the solution you'd like**
While creating a group chat, the user can spawn an emoji picker and choose from those.  
At that point either:
1. the selected emoji gets rasterized and saved as image on the server or
2. the corresponding image for the selected emoji is chosen from a high resolution emoji pack such as https://twemoji.twitter.com/ which also contains SVG versions  

the image is then used as the conversation avatar.

Whatsapp UI:
1. tap the icon button
IMG_8441
2. choose "Emoji and Sticker"
IMG_8442
3. choose emoji and background color
IMG_8443

**Describe alternatives you've considered**
Emoji in the chat title, manually picked picture as a conversation avatar.

**Additional context**
Issue opened as requested by @nickvergessen in #927.

### PR DESCRIPTION
**Pull request by nickvergessen**
### ☑️ Resolves

* Fix #8444 

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | image

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

**Review by jancborchardt — APPROVED** [review]
Looks suuuper cool! :) :sparkles:

File: lib/Service/AvatarService.php [hunk 1]
```diff
@@ -148,6 +137,22 @@ private function getAvatarFolder(string $token): ISimpleFolder {
...
+	 * (0.4 letter-to-total-height ratio, 500*0.4=200), so: 200/0.715 = 280px.
+	 * Since we start from the baseline (text-anchor) we need to
+	 * shift the y axis by 100px (half the caps height): 500/2+100=350
+	 *
+	 * Copied from @see \OC\Avatar\Avatar::$svgTemplate with only `{font}` being modified
+	 */
+	private string $svgTemplate = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+		<svg width="{size}" height="{size}" version="1.1" viewBox="0 0 500 500" xmlns="http://www.w3.org/2000/svg">
+			<rect width="100%" height="100%" fill="#{fill}"></rect>
+			<text x="50%" y="350" style="font-weight:normal;font-size:280px;font-family:{font};text-anchor:middle;fill:#{fgFill}">{letter}</text>
```
Comment (marcoambrosini) [inline lib/Service/AvatarService.php:153]:
```suggestion
			<text x="50%" y="350" style="font-weight:normal;font-size:240px;font-family:{font};text-anchor:middle;fill:#{fgFill}">{letter}</text>
```

Very nice, though some emojis almost touch the boundaries of the avatar circle. I would slightly scale them and give them some air.

**Review by marcoambrosini — APPROVED** [review]

**Review by vitormattos — CHANGES_REQUESTED** [review]

File: lib/Service/AvatarService.php [hunk 2]
```diff
@@ -148,6 +137,22 @@ private function getAvatarFolder(string $token): ISimpleFolder {
...
+		</svg>';
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply lib/Service/AvatarService.php]: That actually generates an error, as there is content before the XML start, etc. so let's stick with the current version.

**Review by vitormattos — APPROVED** [review]

**Comment by vitormattos** [comment]
After merge this PR will be necessary rebase #9423 with this changes and make possible to change the avatar when change the conversation name.

**Comment by SystemKeeper** [comment]
> > when the conversation name is changed, the emoji is not updated:
> 
> It's only the browser caching the image for 24h

Can’t we just get the md5 of the emoji?

**Comment by ShGKme** [comment]
> Can’t we just get the md5 of the emoji?

Do you mean, to calculate `md5` for each conversation everywhere?

**Comment by SystemKeeper** [comment]
> Do you mean, to calculate `md5` for each conversation everywhere?

For cache busting we need a way to identify if an avatar changed. For default avatars the md5 of the filename is used, for custom avatars I think the hash is already at the filename or something, and for emojis this is added at https://github.com/nextcloud/spreed/pull/9423/files#diff-a7e5eb4728a4ad105af312d21ceb1bd6dd7a70068ba67d4b1881099dcae1076bR289-R291

**Pull request merged**